### PR TITLE
Some reviewer commits for https://github.com/sagemath/sage/pull/35682

### PR DESCRIPTION
**Pull request by cheuberg**
### :books: Description

While reading [the code](https://github.com/sagemath/sage/pull/35682), I tried to make some notes on what the various routines are doing; I pushed those notes here; after polishing those, it might be worthwhile to include these.

### :memo: Checklist

- [x] The title is concise, informative, and self-explanatory.
- [x] The description explains in detail what this PR is about.
- [x] I have linked a relevant issue or discussion.
- [ ] I have created tests covering the changes.
- [ ] I have updated the documentation accordingly.

### :hourglass: Dependencies

- #35682: Reviewer commits for this PR.

### CODE REVIEW

File: src/sage/combinat/k_regular_sequence.py [hunk 1]
```diff
@@ -1049,6 +1049,25 @@ def guess(self, f, n_max=100, max_dimension=10, sequence=None):
 
         A :class:`kRegularSequence`
 
+        ALGORITHM:
+
+        For the purposes of this description, the left vector valued sequence
+        associated with a regular sequence is consists of the left vector
```
Comment (dkrenn) [inline src/sage/combinat/k_regular_sequence.py:1055]:
```suggestion
        associated with a regular sequence consists of the left vector
```

File: src/sage/combinat/k_regular_sequence.py [hunk 2]
```diff
@@ -1204,14 +1223,37 @@ def guess(self, f, n_max=100, max_dimension=10, sequence=None):
         zero = domain(0)
         one = domain(1)
 
+        # A `line` will be a triple `(t, r, s)` corresponding to an entry
+        # `k**t * m + r` belonging to `M_s`
+        # TODO: what is `M_s`?
```
Comment (dkrenn) [inline src/sage/combinat/k_regular_sequence.py:1228]:
```suggestion
        # A `line` will be a tuple `(t, r)` corresponding to an entry `k**t * m + r`
```

File: src/sage/combinat/k_regular_sequence.py [hunk 3]
```diff
@@ -1204,14 +1223,37 @@ def guess(self, f, n_max=100, max_dimension=10, sequence=None):
...
+
+        # The elements of `lines` will be correspond to the current components
```
Comment (dkrenn) [inline src/sage/combinat/k_regular_sequence.py:1230]:
```suggestion
        # The elements of `lines` will correspond to the current components
```

File: src/sage/combinat/k_regular_sequence.py [hunk 4]
```diff
@@ -1204,14 +1223,37 @@ def guess(self, f, n_max=100, max_dimension=10, sequence=None):
...
+            """
+                Return current (as defined by ``lines``) left vector valued
+                sequence for argument ``m``.
+            """
             return tuple(seq(m)) + tuple(f(k**t_R * m + r_R) for t_R, r_R, s_R in lines)
 
         @cached_function(key=lambda lines: len(lines))  # we assume that existing lines are not changed (we allow appending of new lines)
         def some_inverse_U_matrix(lines):
+            r"""
+                Find an invertible `d \times d` times submatrix of the matrix
```
Comment (dkrenn) [inline src/sage/combinat/k_regular_sequence.py:1244]:
```suggestion
                Find an invertible `d \times d` submatrix of the matrix
```

File: src/sage/combinat/k_regular_sequence.py [hunk 5]
```diff
@@ -1221,11 +1263,24 @@ def some_inverse_U_matrix(lines):
...
             iU, m_indices = some_inverse_U_matrix(lines)
             X_L = vector(f(k**t_L * m + r_L) for m in m_indices)
             return X_L * iU
 
         def verify_linear_dependence(t_L, r_L, linear_dependence, lines):
+            r"""
+                Determine whether the subsequence with parameters ``t_L`` and
+                ``r_L``, i.e., `m \mapsto f(k**t_L * m + r_L)`, is the linear
+                combination ``linear_dependence`` of the current vector valued
+                sequence.
```
Comment (dkrenn) [inline src/sage/combinat/k_regular_sequence.py:1282]:
```suggestion
                ``r_L``, i.e., `m \mapsto f(k**t_L * m + r_L)`, is the
                ``linear_dependence`` of the current vector valued
                sequence.
```

File: src/sage/combinat/k_regular_sequence.py (hunk 1 above)
Comment (cheuberg) [reply src/sage/combinat/k_regular_sequence.py:1055]: Copied to https://github.com/sagemath/sage/pull/35682/files#r1207823514.

File: src/sage/combinat/k_regular_sequence.py (hunk 2 above)
Comment (cheuberg) [reply src/sage/combinat/k_regular_sequence.py:1228]: Variant posted as https://github.com/sagemath/sage/pull/35682/files#r1207089399.

File: src/sage/combinat/k_regular_sequence.py (hunk 3 above)
Comment (cheuberg) [reply src/sage/combinat/k_regular_sequence.py:1230]: Copied to https://github.com/sagemath/sage/pull/35682/files#r1208409597

File: src/sage/combinat/k_regular_sequence.py (hunk 4 above)
Comment (cheuberg) [reply src/sage/combinat/k_regular_sequence.py:1244]: Copied to https://github.com/sagemath/sage/pull/35682/files#r1208506621 .

File: src/sage/combinat/k_regular_sequence.py (hunk 5 above)
Comment (cheuberg) [reply src/sage/combinat/k_regular_sequence.py:1282]: @dkrenn : I do not think that "linear dependence" is more accurate than "linear dependence".

**Pull request merged**